pkg/gateway/client/tempsetupuser.go

```diff
@@ -0,0 +1,73 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/obot-platform/obot/pkg/gateway/types"
+	"gorm.io/gorm"
+)
+
+// SetTempUserCache stores a temporary user in the database for the bootstrap setup flow.
+// Returns an error if a user is already cached.
+func (c *Client) SetTempUserCache(ctx context.Context, user *types.User, authProviderName, authProviderNS string) error {
+	// Check if a temp user already exists
+	var count int64
+	if err := c.db.WithContext(ctx).Model(&types.TempSetupUser{}).Count(&count).Error; err != nil {
+		return fmt.Errorf("failed to check for existing temp setup user: %w", err)
+	}
+
+	if count > 0 {
+		// Get the existing user's email for the error message
+		var existingUser types.TempSetupUser
+		if err := c.db.WithContext(ctx).First(&existingUser).Error; err != nil {
+			return fmt.Errorf("temporary user already cached")
+		}
+		return fmt.Errorf("temporary user already cached: %s", existingUser.Email)
+	}
+
+	// Create new temp user entry
+	tempUser := &types.TempSetupUser{
+		UserID:           user.ID,
+		Username:         user.Username,
+		Email:            user.Email,
+		Role:             user.Role,
+		IconURL:          user.IconURL,
+		AuthProviderName: authProviderName,
+		AuthProviderNS:   authProviderNS,
+		CreatedAt:        time.Now(),
+	}
+
+	if err := c.db.WithContext(ctx).Create(tempUser).Error; err != nil {
+		return fmt.Errorf("failed to create temp user: %w", err)
+	}
```

Comment:
I think all of this should be in a transaction to avoid race conditions.

Reply:
Fixed